### ISSUE TITLE
Question/Request: Self contained

### ISSUE THREAD

**Issue by Quasozial** [plainionist/TauriDotNetBridge#3]
Hey, i would love to use this tech, but i sadly cant presuppose that all my clients have the .net runtime installed.

Is there any way to deploy the app as self contained or smth, so the runtime doesent have to be installed?

**Comment by RubenPX** [comment]
This should help you https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot

**Comment by Quasozial** [comment]
![image](https://github.com/user-attachments/assets/84090f02-0a1c-4734-af4a-c3d877622b78)
Wouldnt that be a problem, because Dynamic DLL Loading is used here?

**Comment by RubenPX** [comment]
Yes... My bad. I will search for something more. And it's not viable, because tauri need call dotnet methods to load dlls

**Comment by Quasozial** [comment]
yea its probably increasing by more than 100mb. Currently we do that, we ship an WPF Application as self contained with the runtime and its round about 100-200mb

**Comment by RubenPX** [comment]
Yeah... I know... Download .net 9.0 (Desktop only) their size is around 60mb (and it's compressed)

**Comment by Quasozial** [comment]
Hmm, 

i really like this solution here. Because we use as a standard codebase c# but me and my colleague dont like WPF or anything, we are more from the Vue side, so this project is fire for us. I really hope there will be a solution for this

**Comment by RubenPX** [comment]
I don't like wpf too... I my case, personally i like develop with svelte/sveltekit

**Comment by RubenPX** [comment]
I found this... I will investigate this weekend
https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary

**Comment by Quasozial** [comment]
Yea i saw that to after the native aot hint, but it was a bit sketchy for me xD but when it works, it would be fire!

**Comment by RubenPX** [comment]
I tried, but i can't make to work without .net installed...

**Comment by Quasozial** [comment]
hm thats pretty sad, but thanks for your help

**Comment by DonMatron** [comment]
Hi, I think it is possible to install wanted .net runtime during app installation process. If you target OS is Windows, then you can inject .net runtime installer into nsis directly.

It has to be configured in tauri.conf.json:

```
{
  "bundle": {
    "windows": {
      "nsis": {
        "installerHooks": "<path to your .net runtime installation hook>"
      }
    }
  }
}
```

I was not able to create such hook and postponed this issue, since NSIS is pretty tough xD. But my guess is that is possible to use **NSIS_HOOK_PREINSTALL** to install .net runtime before actual app installation process.

**Comment by RubenPX** [comment]
@DonMatron that should be a nice solution, i will investigate more. Maybe nsis can detect if dotnet is installed or better, maybe can detect specific version

**Comment by RubenPX** [comment]
For me i never had problems with antivirus. But one thing that you can make is sign your binary. This avoid toomu h problems with antivirus

https://v2.tauri.app/distribute/sign/windows/

**Comment by plainionist** [comment]
back to AOT: to make this work we would have to compile the plugin and the bridge dll as AOT exe, right? (only then no .NET runtime installation would be required)

and then we would have to check whether https://crates.io/crates/netcorehost is able to load this, right?
